Content.Server/Salvage/SalvageSystem.cs

```diff
@@ -0,0 +1,344 @@
+                    }
+                    else
+                    {
+                        gridState = new SalvageGridState();
+                        gridState.ActiveMagnets.Add(component);
+                        _salvageGridStates[magnetTranform.GridID] = gridState;
+                    }
+                    component.MagnetState = new MagnetState(MagnetStateType.Attaching, gridState.CurrentTime + AttachingTime);
+                    Report("salvage-system-report-activate-success");
+                    break;
+                case MagnetStateType.Attaching:
+                case MagnetStateType.Holding:
+                    ShowPopup("salvage-system-report-already-active", component, user);
+                    break;
+                case MagnetStateType.Detaching:
+                case MagnetStateType.CoolingDown:
+                    ShowPopup("salvage-system-report-cooling-down", component, user);
+                    break;
+                default:
+                    throw new NotImplementedException("Unexpected magnet state type");
+            }
+        }
+        private void ShowPopup(string messageKey, SalvageMagnetComponent component, EntityUid user)
+        {
+            _popupSystem.PopupEntity(Loc.GetString(messageKey), component.Owner, Filter.Entities(user));
+        }
+
+        private void SafeDeleteSalvage(EntityUid salvage)
+        {
+            if(!EntityManager.TryGetComponent<TransformComponent>(salvage, out var salvageTransform))
+            {
+                Logger.ErrorS("salvage", "Salvage entity was missing transform component");
+                return;
+            }
+            //TODO: Figure out how to delete a grid with players, and ghosts on it without crashing anything.
+            foreach(var player in Filter.Empty().AddInGrid(salvageTransform.GridID, EntityManager).Recipients)
+            {
+                if (player.AttachedEntity.HasValue)
+                {
+                    var playerTranform = EntityManager.GetComponent<TransformComponent>(player.AttachedEntity.Value);
+                    playerTranform.AttachParent(salvageTransform.Parent!);
+                }
+            }
+            EntityManager.QueueDeleteEntity(salvage);
+        }
+
+        private bool TryGetSalvagePlacementLocation(out MapCoordinates coords, out Angle angle)
+        {
+            coords = MapCoordinates.Nullspace;
+            angle = Angle.Zero;
+            foreach (var (smc, tsc) in EntityManager.EntityQuery<SalvageMagnetComponent, TransformComponent>(true))
+            {
+                coords = new EntityCoordinates(smc.Owner, smc.Offset).ToMap(EntityManager);
+                var grid = tsc.GridID;
+                if (grid != GridId.Invalid)
+                {
+                    // Has a valid grid - synchronize angle so that salvage doesn't have to deal with cross-grid manipulation issues
+                    angle = _mapManager.GetGrid(grid).WorldRotation;
```

Comment:
Just get WorldRotation through transform.

Reply:
Makes sense, I changed it